cfme/tests/cloud_infra_common/test_power_control_rest.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+import fauxfactory
+import pytest
+
+from cfme.common.vm import VM
+from utils import testgen
+from utils.version import current_version
+from utils.wait import wait_for
+
+
+pytestmark = [
+    pytest.mark.usefixtures('uses_infra_providers', 'uses_cloud_providers')
+]
+
+
+def pytest_generate_tests(metafunc):
+    # Filter out providers without provisioning data or hosts defined
+    argnames, argvalues, idlist = testgen.all_providers(
+        metafunc, 'small_template', template_location=["small_template"])
+
+    testgen.parametrize(metafunc, argnames, argvalues, ids=idlist, scope="module")
+
+
+@pytest.fixture(scope="function")
+def vm_obj(request, provider, setup_provider, small_template, rest_api):
+    vm_obj = VM.factory(
+        'test_pwrctl_{}'.format(fauxfactory.gen_alpha(length=8).lower()),
+        provider, template_name=small_template)
+
+    @request.addfinalizer
+    def _delete_vm():
+        if provider.mgmt.does_vm_exist(vm_obj.name):
+            provider.mgmt.delete_vm(vm_obj.name)
+    vm_obj.create_on_provider(find_in_cfme=True, allow_skip="default")
+    return vm_obj
+
+
+def verify_vm_power_state(vm, state, minutes=20, action=None):
```

Comment:
20 mins because this method is used also when created vm is launching and then powering on + we have rhev provider which loves lock images and the launching procedure takes more time.

We have in cfme/common/vm.py wait_for_vm_state_change method for UI. Should I make verify_vm_power_state also a method of VM? I would prefer to work with power states via rest because we have inconsistency of power state in rest and ui -> https://bugzilla.redhat.com/show_bug.cgi?id=1290003